frontend/util.lua

```diff
@@ -2,9 +2,11 @@
 This module contains miscellaneous helper functions for the KOReader frontend.
 ]]
 
+local ffi = require("ffi")
 local BaseUtil = require("ffi/util")
 local dbg = require("dbg")
 local _ = require("gettext")
+local C = ffi.C
```

Comment:
I think `util.strcoll()` should probably be moved into `base/ffi/util.lua` where all the stuff needing ffi/C is - so that our `frontend/util.lua` keeps being pure Lua and ffi/C free :)
Might be a bit tedious as we need there to add a check for isKobo() and not sure how that would be done. Or check the ffi strcoll is available or not (if that's the reason for it to not be used on Kobo). And possibly get rid of the global DALPHA_SORT_CASE_INSENSITIVE or does it have a purpose? @Frenzie @NiLuJe .

It means making a PR to base and bumping it with this PR (I can do that in a coming up base/crengine bump if that feels too tedious for you).

Reply:
`strcoll` is unusable on Kobo because the system lacks compiled locales.

The check could arguably be made Device-less by checking for the existence of `/usr/lib/locale/locale-archive` instead (c.f., -f /usr/lib/locale/locale-archive).

Reply:
But, regardless, +1 that it belongs in ffi/util, not front's util ;).

Reply:
I'm actually mildly puzzled on *which* of the platforms we target (besides emulator) where `strcoll` actually behaves sanely: it *requires* compiled locales for the current `LC_COLLATE`, something which I highly doubt is happening for !POSIX on eInk readers...

Reply:
And of course, those don't bundle `locale`, so we can't check `locale -a` for the actually compiled locales...

Reply:
On the other hand, we never *actually* set locales to anything other than `en_US.UTF-8`, so just checking for the default locales archive ought to be enough, because if locales are compiled in, there's a good chance that that includes en_US.UTF-8 ;).

Reply:
If it's just to sort dictionary book names, lexical utf8 suffices because this seems like more about some sort of canonical more than anything and unicode point order will "reasonably" sort this alphabetically even in weird scripts.

Heavyweight strcoll stuff is necessary when case gets involved (stricoll), then you need full on accent mark tables.

Reply:
@ezdiy: Yep, this lived in filemanager because it was basically only used by the "Sort By Title" option in the FM until this PR ;).

And, to answer @poire-z's other query: `DALPHA_SORT_CASE_INSENSITIVE` is still enabled by default.